addons/mrp/models/stock_move.py

```diff
@@ -287,7 +290,7 @@ def action_explode(self):
         moves_ids_to_unlink = OrderedSet()
         phantom_moves_vals_list = []
         for move in self:
-            if not move.picking_type_id or (move.production_id and move.production_id.product_id == move.product_id):
+            if self._should_explode_move(move):
```

Comment:
The method name does not really make sense
If it returns `True`, we will actually skip the explode
You could rather call it `_should_skip_explode`
Then, considering my other comment about `self` and `move`, it would become:
```suggestion
            if move._should_skip_explode():
```